extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/scores.html

```diff
@@ -2,89 +2,221 @@
-      <div class="col-1">
-       {diagnosticEntry.key}
-      </div>
-     </div>
-     {#for diagnostic in diagnosticEntry.value}
+{#for endpoint in info:endpointScores.endpoints}
+
+<div class="accordion" id="accordionExample">
+  <div class="card">
+
+    <div class="card-header" id="headingOne">
       <div class="row">
-       <div class="col-1"></div>
-       <div class="col-1"></div>
-       <div class="col">
-        {diagnostic.message}
-       </div>
-       <div class="col-1">
-        <span class="badge badge-{#if diagnostic.score < 50}danger{#else if diagnostic.score < 100}warning{#else}success{/if}">{diagnostic.score}/100</span>
-       </div>
-      </div>
-     {/for}
-    {/for}
-    <div class="row">
-     <div class="col-1"></div>
-      <div class="col-1">
-        Filters
+        <div class="col">
+          <h2 class="mb-0">
+            <button class="btn btn-block text-left collapsed" type="button" data-toggle="collapse"
+                    data-target=#{endpoint.httpMethod} aria-expanded="false"
+                    aria-controls={endpoint.httpMethod}>
+              <code>{endpoint.httpMethod} {endpoint.fullPath}</code> <b class="score"> {endpoint.score}%</b>
+<!--              <span class="badge badge-{#if endpoint.score < 50}danger{#else if endpoint.score < 100}warning{#else}success{/if}">{endpoint.score}/100</span>-->
+            </button>
+          </h2>
+        </div>
       </div>
     </div>
-    <div class="row">
-      <div class="col-1"></div>
-      <div class="col-1"></div>
-      <div class="col">
-        <span class="badge badge-light" id="collapse-list" data-toggle="collapse" data-target="#filter">
-          <i class="fas fa-greater-than"></i>
-        </span>
-      </div>
+
+    <div id={endpoint.httpMethod} class="collapse" aria-labelledby="headingOne"
+         data-parent="#accordionExample">
+      <div class="card-body">
+
+
+        <div class="row" style="margin-left:2%;">
+        {#for diagnosticEntry in endpoint.diagnostics.entrySet()}
+          {#for diagnostic in diagnosticEntry.value}
+        <div class="col-4">
+
+        <div class="row">
+        <div class="gauge">
+          <div class="gauge__body">
+            <div class="gauge__fill"></div>
+            <div class="gauge__cover">{diagnostic.score}%</div>
+          </div>
+        </div>
+        </div>
+
+
+        <div class="row">
+          <div class="key_style">
+          {diagnosticEntry.key}
+          </div>
+        </div>
+
+        {| <script>
+          const gaugeElement = document.querySelector(".gauge");
+
+          function setGaugeValue(gauge, value) {
+          if (value < 0 || value > 1) {
+            return;
+          }
+
+          gauge.querySelector(".gauge__fill").style.transform = `rotate(${value / 2}turn)`;
+          gauge.querySelector(".gauge__cover").textContent = `${Math.round(value * 100)}%`;
+        }
+
+        setGaugeValue(gaugeElement, \{diagnostic.score}/100);
```

Comment:
This line @mkouba
Replacing `{diagnostic.score}` with numeric value is working fine.

Reply:
Ok, so you can either remove the unparsed character data section, i.e. `{|` and `|}` and escape the first bracket for `{value / 2}` and `{Math.round(value * 100)}` - we need to ignore these during rendering:

```javascript
  const gaugeElement = document.querySelector(".gauge");
  function setGaugeValue(gauge, value) {
    if (value < 0 || value > 1) {
      return;
    }
    gauge.querySelector(".gauge__fill").style.transform = `rotate($\{value / 2}turn)`;
    gauge.querySelector(".gauge__cover").textContent = `$\{Math.round(value * 100)}%`;
  }
  setGaugeValue(gaugeElement, {diagnostic.score}/100);
```
Alternatively, you can end the unparsed character data section right before the `{diagnostic.score}` expr. And of course, you can't escape the `{diagnostic.score}` expr because you need to output the value (not ignore ;-).

```javascript
{| <script>
  const gaugeElement = document.querySelector(".gauge");
  function setGaugeValue(gauge, value) {
   if (value < 0 || value > 1) {
     return;
   }
  gauge.querySelector(".gauge__fill").style.transform = `rotate(${value / 2}turn)`;
  gauge.querySelector(".gauge__cover").textContent = `${Math.round(value * 100)}%`;
  }
|}
setGaugeValue(gaugeElement, {diagnostic.score}/100);
```

BTW I noticed several JS errors in the console:
`Uncaught SyntaxError: redeclaration of const gaugeElement
    <anonymous> http://localhost:8080/q/dev/io.quarkus.quarkus-resteasy-reactive/scores:247`

Reply:
And FYI we now throw an exception if a property is not found in an expression, just rebase this PR on master ;-).

Reply:
perfect this is working
I did the alternative way you suggested :)

> Uncaught SyntaxError: redeclaration of const gaugeElement <anonymous> http://localhost:8080/q/dev/io.quarkus.quarkus-resteasy-reactive/scores:247

Where you caught this, it's not showing on running `mvn quarkus:dev` nor in logs. 

But yes I got the point, the JS snippet is in `for` loop and for each `endpoint` it is getting declared and do you know the JS snippet is working only for 1st time. after that I think it stops due to `redeclaration of const gaugeElement`

I think I need to declare `const gaugeElement = document.querySelector(".gauge");` globally somewhere

Reply:
> Where you caught this, it's not showing on running mvn quarkus:dev nor in logs.

In the browser console, it's a JS error ;-)

Reply:
Instead of `const`, made `gaugeElement` a variable and no issues in Console now :)
![image](https://user-images.githubusercontent.com/46818757/110111377-3d4f1b80-7dd6-11eb-837c-07120f8b6193.png)